docs/designs/xref.md

```diff
@@ -0,0 +1,128 @@
+# Xref
+Besides using file path to link to another file, DocFX also allows you to give a file a unique identifier so that you can reference this file using that identifier instead of its file path. This is useful in the following cases:
+- Path to file is long and difficult to memorize or changes frequently.
+- API reference documentation is usually auto generated so it's difficult to find its file path.
+- You want to reference to files in another project without need to know its file structure.
+
+## Define UID
+The unique identifier of a file is called UID (stands for unique identifier) in DocFX.
+- User can define UID in YAML header of markdown file
+```
+---
+uid: a
+title: ASP.NET Documentation
+description: Learn how to develop ASP.NET and ASP.NET web applications. Get documentation, example code, tutorials, and more.
+---
+```
+- User can also define UID in JSON
+```json
+{
+  "$schema": "https://raw.githubusercontent.com/dotnet/docfx/v3/schemas/TestData.json",
+  "uid": "a",
+  "title": "ASP.NET Documentation",
+  "description": "Learn how to develop ASP.NET and ASP.NET web applications. Get documentation, example code, tutorials, and more."
+}
+```
+- And define UID in YAML
+```yml
```

Comment:
It'd be better to mention that it is the SDP schema that defines where we can specify `uid` in Yaml or Json files.